k8s/envoy-jwt/README.md

```diff
@@ -0,0 +1,329 @@
+
+Intentionally, there is no registration entry for the frontend-2 service. It will be added later to demonstrate that requests are denied by the external authorization filter when a JWT-SVID is not present in the request header.
+
+
+# Part 2: Test Connections
+
+Now that services are deployed and also registered in SPIRE, let's test the authorization that we've configured.
+
+## Testing for Valid and Invalid JWT-SVIDs
+The first set of testing will demonstrate how valid JWT-SVIDs allow for the display of associated data and invalid JWT-SVIDs prevent the associated data from being displayed. To run these tests, we need to find the IP addresses and ports that make up the URLs to use for accessing the data.
+
+   ```console
+   $ kubectl get services
+
+   NAME            TYPE           CLUSTER-IP    EXTERNAL-IP      PORT(S)          AGE
+   backend-envoy   ClusterIP      None          <none>           9001/TCP         6m53s
+   frontend        LoadBalancer   10.8.14.117   35.222.164.221   3000:32586/TCP   6m52s
+   frontend-2      LoadBalancer   10.8.7.57     35.222.190.182   3002:32056/TCP   6m53s
+   kubernetes      ClusterIP      10.8.0.1      <none>           443/TCP          59m
+   ```
+
+The `frontend` service will be available at `EXTERNAL-IP` value and port `3000`, which was configured for our container. In the sample output shown above, the URL to navigate is `35.222.164.221:3000`. Open your browser and navigate to the IP address shown for `frontend` in your environment, adding the port `:3000`. Once the page is loaded, you'll see the account details for user _Jacob Marley_. 
+
+![Frontend][frontend-view]
+
+[frontend-view]: images/frontend_view.png "Frontend view"
+
+On the other hand, following the same steps we can confirm that the `frontend-2` service is not able to talk to the `backend` service. This is because it was not updated to include a JWT token in the request. The lack of a valid token on the request makes the Envoy instance in front of the `backend` reject it.
+As a result, when you connect to the URL for the `frontend-2` service (e.g. `35.222.190.182:3002`), the browser only displays the title without any account details.
+
+![Frontend-2-no-details][frontend-2-view-no-details]
+
+[frontend-2-view-no-details]: images/frontend-2_view_no_details.png "Frontend-2 view no details account"
+
+Let's take a look at the `auth-helper` container logs to see what is happening behind the scenes. The following are the logs for the `auth-helper` instance running next to the `frontend` service. In this case, the `auth-helper` server is configured to run in inject mode. For every request, it will inject the JWT-SVID as a new request header and return it to the Envoy instance that will forward it to the `backend`.
+
+```console
+   $ kubectl logs -f --selector=app=frontend -c auth-helper
+   Envoy JTW Auth Helper running in jwt_injection mode
```

Comment:
JTW instead of JWT here and below. Probably need to update the code.

Reply:
Yes, the original code need to be updated. Good news is that the previous change is not pushed yet :D